### PR TITLE
Deprecate the extending of containers

### PR DESCRIPTION
**Pull request by andydotxyz**
This shouldn't really have been supported.
Relates to the accidental breaking of Scroll extending - discussed on call 5th Feb.

### Checklist:

- [ ] Tests included. <- just deprecations
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: container/apptabs.go [hunk 1]
```diff
@@ -106,9 +106,16 @@ func (c *AppTabs) CurrentTabIndex() int {
 	return c.current
 }
 
+// ExtendBaseWidget is used by an extending widget to make use of BaseWidget functionality.
+//
+// Deprecated: Support for extending containers is being removed
+func (c *AppTabs) ExtendBaseWidget(wid fyne.Widget) {
```
Comment (Jacalz) [inline container/apptabs.go:112]: I thought we said, on the call yesterday, that we wouldn't add "new" deprecated methods?

File: container/split.go [hunk 2]
```diff
@@ -60,6 +60,13 @@ func (s *Split) CreateRenderer() fyne.WidgetRenderer {
 	}
 }
 
+// ExtendBaseWidget is used by an extending widget to make use of BaseWidget functionality.
+//
+// Deprecated: Support for extending containers is being removed
+func (s *Split) ExtendBaseWidget(wid fyne.Widget) {
```
Comment (Jacalz) [inline container/split.go:66]: I thought we said, on the call yesterday, that we wouldn't add "new" deprecated methods?

**Review by Jacalz — COMMENTED** [review]
Just a question regarding the implementation which I thought we said something else about on the call yesterday. I am not technically against it, just asking. Aparart from my question, it looks good 👍

File: container/apptabs.go (hunk 1 above)
Comment (andydotxyz) [reply container/apptabs.go:112]: Yes we did. I have not added the new no-op `ExtendBaseWidget` in `Scroll` where it would be required.
These methods are not "new" they are wrappers so that I can mark Deprecated.
Obviously we don't want to set it Deprecated on `widget.BaseWidget` which is what it overrides now.

Comment (Jacalz) [reply container/apptabs.go:112]: Ah, I see. Thanks for the clarification.

**Review by Jacalz — APPROVED** [review]

File: container/apptabs.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func NewAppTabs(items ...*TabItem) *AppTabs {
 		// Current is first tab item
 		tabs.current = 0
 	}
-	tabs.ExtendBaseWidget(tabs)
+	tabs.BaseWidget.ExtendBaseWidget(tabs)
```
Comment (stuartmscott) [inline container/apptabs.go:59]: As the Deprecated ExtendBaseWidget is not a noop, are these changes needed?

Comment (andydotxyz) [reply container/apptabs.go:59]: With it this change you have deprecation warnings... that seemed undesirable

Comment (stuartmscott) [reply container/apptabs.go:59]: Ah fair enough

**Review by stuartmscott — APPROVED** [review]
